assets/js/core/WakaTimeCore.js

```diff
@@ -117,8 +114,9 @@ class WakaTimeCore {
                             }
 
                             if (items.loggingStyle == 'whitelist') {
-                                if (contains(currentActiveTab.url, items.whitelist)) {
-                                    this.sendHeartbeat(currentActiveTab.url, debug);
+                                entity = this.checkURL(currentActiveTab.url, items.whitelist);
```

Comment:
Change this line to:

`heartbeat = this.getHeartbeat(currentActiveTab.url, items.whitelist);`